pkg/api/customization/logging/action.go

```diff
@@ -166,9 +174,9 @@ func (h *Handler) dryRunLoggingTarget(apiContext *types.APIContext, level, clust
 		return err
 	}
 
-	podLister := context.Core.Pods(loggingconfig.LoggingNamespace).Controller().Lister()
+	pods := context.Core.Pods(loggingconfig.LoggingNamespace)
```

Comment:
This should continue to use a lister, if it's not being initialized then the initialization can be moved to ensure the cache is primed once we get here.

Reply:
Found that 2.4 is different with master, in 2.4 we don't start the controller in userContext here https://github.com/rancher/rancher/blob/release/v2.4/pkg/clustermanager/manager.go#L428, so we can't use lister here, no sure we can simply start the controller or need more changes.

Reply:
That link goes to where we do start the controller, is there another issue being seen?

Reply:
Yes, the start function passes false to the third parameter, so I can't get pods by Lister from the userContext.
```
func (m *Manager) start(ctx context.Context, cluster *v3.Cluster, controllers, clusterOwner bool) (*record, error) 
```

Reply:
Expand `RegisterFollower` to include the resource needed, that will allow the cache to be ready.

Reply:
Updated, thank you